temperature/router.py

```diff
@@ -0,0 +1,35 @@
+from fastapi import APIRouter, Depends, Query, HTTPException
+from sqlalchemy.orm import Session
+
+from dependencies import get_db
+from temperature import crud, schemas
+
+
+router = APIRouter()
+
+
+@router.get(
+    "/temperature/",
+    response_model=list[schemas.Temperature]
+)
+async def get_temperature_list(
+        city_id: int = Query(None, description="City id"),
+        db: Session = Depends(get_db)
```

Comment:
The `db` parameter is defined as a `Session`, but the CRUD operations expect an `AsyncSession`. Ensure that `get_db` provides an `AsyncSession` to avoid runtime errors.